builder/virtualbox/iso/step_create_vm.go

```diff
@@ -26,58 +26,56 @@ func (s *stepCreateVM) Run(ctx context.Context, state multistep.StateBag) multis
-	commands[1] = []string{
+	})
+	commands = append(commands, []string{
 		"modifyvm", name,
 		"--boot1", "disk", "--boot2", "dvd", "--boot3", "none", "--boot4", "none",
-	}
-	commands[2] = []string{"modifyvm", name, "--cpus", strconv.Itoa(config.HWConfig.CpuCount)}
-	commands[3] = []string{"modifyvm", name, "--memory", strconv.Itoa(config.HWConfig.MemorySize)}
-	commands[4] = []string{"modifyvm", name, "--usb", map[bool]string{true: "on", false: "off"}[config.HWConfig.USB]}
+	})
+	commands = append(commands, []string{"modifyvm", name, "--cpus", strconv.Itoa(config.HWConfig.CpuCount)})
+	commands = append(commands, []string{"modifyvm", name, "--memory", strconv.Itoa(config.HWConfig.MemorySize)})
+	commands = append(commands, []string{"modifyvm", name, "--usb", map[bool]string{true: "on", false: "off"}[config.HWConfig.USB]})
 
 	if strings.ToLower(config.HWConfig.Sound) == "none" {
-		commands[5] = []string{"modifyvm", name, "--audio", config.HWConfig.Sound,
-			"--audiocontroller", config.AudioController}
+		commands = append(commands, []string{"modifyvm", name, "--audio", config.HWConfig.Sound,
+			"--audiocontroller", config.AudioController})
 	} else {
-		commands[5] = []string{"modifyvm", name, "--audio", config.HWConfig.Sound, "--audioin", "on", "--audioout", "on",
-			"--audiocontroller", config.AudioController}
+		commands = append(commands, []string{"modifyvm", name, "--audio", config.HWConfig.Sound, "--audioin", "on", "--audioout", "on",
+			"--audiocontroller", config.AudioController})
 	}
 
-	commands[6] = []string{"modifyvm", name, "--chipset", config.Chipset}
-	commands[7] = []string{"modifyvm", name, "--firmware", config.Firmware}
+	commands = append(commands, []string{"modifyvm", name, "--chipset", config.Chipset})
+	commands = append(commands, []string{"modifyvm", name, "--firmware", config.Firmware})
 	// Set the configured NIC type for all 8 possible NICs
-	commands[8] = []string{"modifyvm", name,
+	commands = append(commands, []string{"modifyvm", name,
 		"--nictype1", config.NICType,
 		"--nictype2", config.NICType,
 		"--nictype3", config.NICType,
 		"--nictype4", config.NICType,
 		"--nictype5", config.NICType,
 		"--nictype6", config.NICType,
 		"--nictype7", config.NICType,
-		"--nictype8", config.NICType}
-	commands[9] = []string{"modifyvm", name, "--graphicscontroller", config.GfxController, "--vram", strconv.FormatUint(uint64(config.GfxVramSize), 10)}
+		"--nictype8", config.NICType})
+	commands = append(commands, []string{"modifyvm", name, "--graphicscontroller", config.GfxController, "--vram", strconv.FormatUint(uint64(config.GfxVramSize), 10)})
 	if config.RTCTimeBase == "UTC" {
-		commands[10] = []string{"modifyvm", name, "--rtcuseutc", "on"}
+		commands = append(commands, []string{"modifyvm", name, "--rtcuseutc", "on"})
 	} else {
-		commands[10] = []string{"modifyvm", name, "--rtcuseutc", "off"}
+		commands = append(commands, []string{"modifyvm", name, "--rtcuseutc", "off"})
 	}
 	if config.NestedVirt == true {
```

Comment:
Should `NestedVirt` be a trilean ? And when not set we use the old default otherwise use the actual value ?

Reply:
I think it makes sense. I'm not sure if the default is always off 🤔  better to think is not